astroquery/jplhorizons/core.py

```diff
@@ -505,36 +736,43 @@ def elements_async(self, get_query_payload=False,
             if no_fragments:
                 commandline += ' NOFRAG;'
 
+        if isinstance(self.location, dict):
+            raise ValueError(('cannot use topographic position in orbital'
+                              'elements query'))
+
         # configure request_payload for ephemerides query
         request_payload = OrderedDict([
             ('batch', 1),
             ('TABLE_TYPE', 'ELEMENTS'),
+            ('MAKE_EPHEM', 'YES'),
             ('OUT_UNITS', 'AU-D'),
             ('COMMAND', '"' + commandline + '"'),
             ('CENTER', ("'" + str(self.location) + "'")),
-            ('CSV_FORMAT', ('"YES"')),
-            ('REF_PLANE', 'ECLIPTIC'),
-            ('REF_SYSTEM', 'J2000'),
-            ('TP_TYPE', 'ABSOLUTE'),
+            ('CSV_FORMAT', 'YES'),
             ('ELEM_LABELS', 'YES'),
-            ('OBJ_DATA', 'YES')]
+            ('OBJ_DATA', 'YES'),
+            ('REF_SYSTEM', refsystem),
+            ('REF_PLANE', {'ecliptic': 'ECLIPTIC', 'earth': 'FRAME',
+                           'body': "'BODY EQUATOR'"}[refplane]),
+            ('TP_TYPE', {'absolute': 'ABSOLUTE',
+                         'relative': 'RELATIVE'}[tp_type])]
         )
 
         # parse self.epochs
         if isinstance(self.epochs, (list, tuple, ndarray)):
-            request_payload['TLIST'] = "\n".join([str(epoch) for epoch in
+            request_payload['TLIST'] = "\n".join([str(epoch) for
+                                                  epoch in
                                                   self.epochs])
         elif type(self.epochs) is dict:
             if ('start' not in self.epochs or 'stop' not in self.epochs or
                     'step' not in self.epochs):
-                raise ValueError("'epochs' parameter must contain start, " +
+                raise ValueError("'epochs' must contain start, " +
```

Comment:
no need for the `+` at the end of the line

Reply:
done